apps/spruce/src/pages/task/taskTabs/TaskHistory/caching/index.ts

```diff
@@ -0,0 +1,170 @@
+import { FieldMergeFunction, FieldReadFunction } from "@apollo/client";
+import { ReadFieldFunction } from "@apollo/client/cache/core/types/common";
+import { TaskHistoryQuery, TaskHistoryDirection } from "gql/generated/types";
+import { ACTIVATED_TASKS_LIMIT } from "../constants";
+import { TaskHistoryTask, TaskHistoryPagination } from "../types";
+
+type TaskHistoryCache = TaskHistoryQuery["taskHistory"] & {
+  allTaskOrders?: Set<number>;
+};
+
+export const isAllInactive = (
+  tasks: TaskHistoryTask[],
+  readField: ReadFieldFunction,
+) => {
+  for (let i = 0; i < tasks.length; i++) {
+    const activated = readField<boolean>("activated", tasks[i]) ?? false;
+    if (activated) {
+      return false;
+    }
+  }
+  return true;
+};
```

Comment:
nit you can probably just rewrite this to
```suggestion
): boolean => tasks.every((t) => !readField<boolean>("activated", t));
```